### PR TITLE
feat: add `IfExpr` class

### LINKED ISSUE #93: `IfExpr` - Represents an if expression (e.g., in Rust).
suggestion from gpt:

```python

# Add this code to src/astx/datatypes.py

from typing import Optional
from astx.base import (
    NO_SOURCE_LOCATION,
    ASTKind,
    DataTypeOps,
    ExprType,
    SourceLocation,
    ASTNodes,
)
from astx.types import ReprStruct
from public import public

@public
class IfExpr(DataTypeOps):
    """AST class for the 'if' expression."""

    condition: DataTypeOps
    then_expr: DataTypeOps
    else_expr: DataTypeOps
    type_: ExprType

    def __init__(
        self,
        condition: DataTypeOps,
        then_expr: DataTypeOps,
        else_expr: DataTypeOps,
        loc: SourceLocation = NO_SOURCE_LOCATION,
        parent: Optional[ASTNodes] = None,
    ) -> None:
        """Initialize the IfExpr instance."""
        super().__init__(loc=loc, parent=parent)
        self.condition = condition
        self.then_expr = then_expr
        self.else_expr = else_expr
        self.kind = ASTKind.IfExprKind  # We'll add this to ASTKind
        self.type_ = self._infer_type()
        self.loc = loc
        self.parent = parent

    def _infer_type(self) -> ExprType:
        """Infer the resulting type of the IfExpr."""
        if self.then_expr.type_ == self.else_expr.type_:
            return self.then_expr.type_
        else:
            # Implement logic to determine the resulting type
            # For simplicity, we can choose a common ancestor or promote types
            return AnyExpr

    def __str__(self) -> str:
        """Return a string that represents the object."""
        return f"IfExpr({self.condition}, {self.then_expr}, {self.else_expr})"

    def get_struct(self, simplified: bool = False) -> ReprStruct:
        """Return the AST structure of the object."""
        key = "IfExpr"
        value = {
            "condition": self.condition.get_struct(simplified),
            "then": self.then_expr.get_struct(simplified),
            "else": self.else_expr.get_struct(simplified),
        }
        return self._prepare_struct(key, value, simplified)
```

### PR DESCRIPTION
**Pull request by apkrelling**
## Pull Request description

This PR adds `IfExpr` and modifies `If` -> `IfStmt`. 
This PR is an attempt to resolve #93  .

- [X] added new ASTKind
- [X] created `IfExpr` class
- [X] updated `__init__.py` to include the new imports and classes in the `__all__` list
- [X] added test for `IfExpr`
- [X] added transpiler visit method for `IfExpr`
- [X] added transpiler test for `IfExpr`
- [X] added transpiler visit method for `IfStmt`
- [X] added transpiler tests for `IfStmt`

## How to test these changes
```python
from astx.operators import BinaryOp
from astx.blocks import Block
from astx.datatypes import LiteralInt32
from astx.flows import IfExpr
from astx.transpilers import python as astx2py

# determine condition
cond = BinaryOp(op_code=">", lhs=LiteralInt32(1), rhs=LiteralInt32(2))

# create then and else blocks
then_block = Block()
else_block = Block()

# define literals
lit_2 = LiteralInt32(2)
lit_3 = LiteralInt32(3)

# define operations
op1 = lit_2 + lit_3
op2 = lit_2 - lit_3

# Add statements to the then and else blocks
then_block.append(op1)
else_block.append(op2)

# define if Expr
if_expr = IfExpr(condition=cond, then=then_block, else_=else_block)

if_expr
if_expr.__str__()

# Initialize the generator
generator = astx2py.ASTxPythonTranspiler()

# Generate Python code
generated_code = generator.visit(if_expr)
generated_code
```
if_expr_ascii
if_expr_png

Output of `if_expr.__str__()`: `'IfExpr[BinaryOp[>](LiteralInt32(1),LiteralInt32(2))]'`

Output of `generated_code`: `'result =     (2 + 3) if  (1 > 2) else     (2 - 3)'`
## Pull Request checklists

This PR is a:

- [ ] bug-fix
- [X] new feature
- [ ] maintenance

About this PR:

- [X] it includes tests.
- [X] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [X] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [X] I have reviewed the changes and it contains no misspelling.
- [X] The code is well commented, especially in the parts that contain more
      complexity.
- [X] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

**Comment by xmnlab** [comment]
hi @apkrelling,

about the output from the transpiler for the `if expression`, python has already this syntax, although it is for an inline expression.

maybe you can use this as a reference:

```
>>> import ast
>>> ast_text = "a = 1 if mycond == 1 else 2 if mycond == 2 else 3"
>>> pprint(ast.dump(ast.parse(ast_text)))
("Module(body=[Assign(targets=[Name(id='a', ctx=Store())], "
 "value=IfExp(test=Compare(left=Name(id='mycond', ctx=Load()), ops=[Eq()], "
 'comparators=[Constant(value=1)]), body=Constant(value=1), '
 "orelse=IfExp(test=Compare(left=Name(id='mycond', ctx=Load()), ops=[Eq()], "
 'comparators=[Constant(value=2)]), body=Constant(value=2), '
 'orelse=Constant(value=3))))], type_ignores=[])')
>>> 

```

**Comment by xmnlab** [comment]
for if stmt, the result from the transpiler would look like this:

```python

if  (1 > 2):
    x = (2 + 3)
else:
    x = (2 - 3)'
```

File: src/astx/transpilers/python.py [hunk 1]
```diff
@@ -81,6 +81,21 @@ def visit(self, node: astx.ForRangeLoopExpr) -> str:
             f"{self.visit(node.step)})]"
         )
 
+    @dispatch  # type: ignore[no-redef]
+    def visit(self, node: astx.IfExpr) -> str:
+        """Handle IfExpr nodes."""
+        if node.else_:
+            return (
+                f"result = {self.visit(node.then)} if "
```
Comment (xmnlab) [inline src/astx/transpilers/python.py:89]:
```suggestion
                f"{self.visit(node.then)} if "
```

File: src/astx/transpilers/python.py [hunk 2]
```diff
@@ -81,6 +81,21 @@ def visit(self, node: astx.ForRangeLoopExpr) -> str:
...
+                f" {self.visit(node.condition)}"
+                f" else {self.visit(node.else_)}"
+            )
+
+        return (
+            f"result = {self.visit(node.then)} if "
```
Comment (xmnlab) [inline src/astx/transpilers/python.py:95]:
```suggestion
            f"{self.visit(node.then)} if "
```

**Comment by apkrelling** [comment]
Transpiler outputs:
```python
from astx.operators import BinaryOp
from astx.blocks import Block
from astx.datatypes import LiteralInt32
from astx.flows import IfExpr, IfStmt
from astx.transpilers import python as astx2py

# determine condition
cond = BinaryOp(op_code=">", lhs=LiteralInt32(1), rhs=LiteralInt32(2))

# create then and else blocks
then_block = Block()
else_block = Block()

# define literals
lit_2 = LiteralInt32(2)
lit_3 = LiteralInt32(3)

# define operations
op1 = lit_2 + lit_3
op2 = lit_2 - lit_3

# Add statements to the then and else blocks
then_block.append(op1)
else_block.append(op2)

# define if Stmt/Expr
if_stmt1 = IfStmt(condition=cond, then=then_block, else_=else_block)
if_stmt2 = IfStmt(condition=cond, then=then_block)
if_expr1 = IfExpr(condition=cond, then=then_block, else_=else_block)
if_expr2 = IfExpr(condition=cond, then=then_block)

# Initialize the generator
generator = astx2py.ASTxPythonTranspiler()

# Generate Python code
generated_code_stmt1 = generator.visit(if_stmt1)
generated_code_stmt2 = generator.visit(if_stmt2)
generated_code_expr1 = generator.visit(if_expr1)
generated_code_expr2 = generator.visit(if_expr2)

print(generated_code_stmt1)
print(generated_code_stmt2)
print(generated_code_expr1)
print(generated_code_expr2)
```

Output of `print(generated_code_stmt1)`: 
`if (1 > 2):
    (2 + 3)
else:
    (2 - 3)
`
Output of `print(generated_code_stmt2)`:
if (1 > 2):
    (2 + 3)
Output of `print(generated_code_expr1)`:
`    (2 + 3) if  (1 > 2) else     (2 - 3)`
Output of `print(generated_code_expr2)`:
`    (2 + 3) if  (1 > 2)`

File: tests/transpilers/test_python.py [hunk 3]
```diff
@@ -599,3 +639,117 @@ def test_transpiler_while_expr() -> None:
...
+    if_expr = astx.IfExpr(condition=cond, then=then_block)
+
+    # Initialize the generator
+    generator = astx2py.ASTxPythonTranspiler()
+
+    # Generate Python code
+    generated_code = generator.visit(if_expr)
+
+    # Expected code for the binary operation
+    expected_code = "    (2 + 3) if  (1 > 2)"
```
Comment (xmnlab) [inline tests/transpilers/test_python.py:674]: in the case when there is no "else", it should return "None"

something like this:

```suggestion
    expected_code = "    (2 + 3) if  (1 > 2) else None"
```

**Comment by apkrelling** [comment]
I updated the transpiler method for `IfExpr`, and now the outputs are:

```
Output of `print(generated_code_expr1)`:
```
    (2 + 3) if  (1 > 2) else     (2 - 3)
```
Output of `print(generated_code_expr2)`:
```
    (2 + 3) if  (1 > 2) else None
```

**Pull request merged**